StateNet/Network/NetworkBuilder.cs

```diff
@@ -127,23 +128,30 @@ public NetworkBuilder ClearConnectionsWithInput(string input)
 
         private StateNetworkResult CreateStateNetwork()
         {
+
+            
             try
             {
                 var stateDictionary =
-                    new Dictionary<string, IReadOnlyDictionary<string, IReadOnlyList<Connection>>>();
+                new Dictionary<string, IReadOnlyDictionary<string, IReadOnlyList<Connection>>>();
 
                 foreach (var state in _states)
                 {
                     var inputDictionary = new Dictionary<string, IReadOnlyList<Connection>>();
 
+                    foreach (var input in _inputs)
+                    {
+                        inputDictionary.Add(input, new List<Connection>()); //This requires every input to have at least an empty connection associated to it
```

Comment:
Good idea, we will have to make sure this behavior is reflected in the tests & engine